pyte/screens.py

```diff
@@ -1043,6 +1048,26 @@ def debug(self, *args, **kwargs):
         By default is a noop.
         """
 
+    def scroll_up(self, lines):
+        """Scroll up `lines` lines."""
+        lines_to_scroll = range(self.margins.top, self.margins.bottom + 1)
+        for y in lines_to_scroll:
+            if y + lines in set(lines_to_scroll):
+                self.buffer[y] = self.buffer[y + lines].copy()
+            else:
+                self.buffer[y].clear()
+        self.dirty = set(lines_to_scroll)
+
+    def scroll_down(self, lines):
+        """Scroll down `lines` lines."""
+        lines_to_scroll = range(self.margins.bottom, self.margins.top - 1, -1)
+        for y in lines_to_scroll:
+            if y - lines in set(lines_to_scroll):
+                self.buffer[y] = self.buffer[y - lines].copy()
+            else:
+                self.buffer[y].clear()
+        self.dirty = set(lines_to_scroll)
```

Comment:
Similar to scroll_up, this should also default to 1 line if the argument is unspecified. However, an argument of 0 is used to reset mouse tracking (may be xterm specific). In any case, if we get an argument of 0, we don't want to scroll at all. And if we don't get an argument we do want to scroll exactly 1 line. However, looking at the current implementation of `_parser_fsm`, we may be unable to differentiate between receiving no arguments and receiving a 0 as the first argument.

Any other changes made to scroll_up should be appropriately mirrored here in scroll_down.